### PR TITLE
SPLAT-828: Modify vSphere platform specification to include network configuration

### PR DESCRIPTION
**Pull request by vr4manta**
[SPLAT-828](https://issues.redhat.com//browse/SPLAT-828): Enhanced installer to have network configuration in the vsphere platform specification so that static IPs can be configured.

### CODE REVIEW

File: pkg/types/vsphere/validation/platform_test.go [hunk 1]
```diff
@@ -57,6 +57,62 @@ func validPlatform() *vsphere.Platform {
 	}
 }
 
+func validHosts() []*vsphere.Host {
```
Comment (rvanderp3) [inline pkg/types/vsphere/validation/platform_test.go:60]:
```suggestion
// ValidHosts returns a slice of host configurations for use in other test suites
func ValidHosts() []*vsphere.Host {
```
 
I found a need for this in the terraform unit tests

Comment (rvanderp3) [reply pkg/types/vsphere/validation/platform_test.go:60]: hmmm ... i thought I export cross package but that doesn't seem to be working for me.  for now, ignore.

File: pkg/types/vsphere/validation/platform.go [hunk 2]
```diff
@@ -221,3 +223,100 @@ func validateLoadBalancer(lbType configv1.PlatformLoadBalancerType) bool {
...
+// validateIP checks IP value to see if it is valid.  If IP is required, an error will be returned if
+// the IP is not specified.
+func validateIP(ip string, req bool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if ip == "" && req {
+		allErrs = append(allErrs, field.Required(fldPath, "must specify a IP"))
+	} else if ip != "" {
+		if valErr := validate.IP(ip); valErr != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, ip, valErr.Error()))
+		}
```
Comment (rvanderp3) [inline pkg/types/vsphere/validation/platform.go:319]: The [CAPV API spec](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/docs/troubleshooting.md#a-static-ip-address-must-include-the-segment-length) is ambiguous and yesterday I was questioning how the CIDR was communicated.  Turns out I we have to [provide the CIDR](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/docs/troubleshooting.md#a-static-ip-address-must-include-the-segment-length) as well.

Comment (rvanderp3) [reply pkg/types/vsphere/validation/platform.go:319]: Unit tested with the addition of this function for the validation of the host IP address:
`
// validateIPWithCidr checks IP/CIDR value to see if it is valid.  If IP is required, an error will be returned if
// the IP is not specified.
func validateIPWithCidr(ip string, req bool, fldPath *field.Path) field.ErrorList {
	allErrs := field.ErrorList{}
	if ip == "" && req {
		allErrs = append(allErrs, field.Required(fldPath, "must specify a IP address with CIDR"))
	} else if ip != "" {
		if _, _, valErr := net.ParseCIDR(ip); valErr != nil {
			allErrs = append(allErrs, field.Invalid(fldPath, ip, valErr.Error()))
		}
	}
	return allErrs
}
`

Comment (vr4manta) [reply pkg/types/vsphere/validation/platform.go:319]: Ok.  Updated API / functions to accept CIDR.

File: pkg/types/vsphere/validation/platform_test.go (hunk 1 above)
Comment (vr4manta) [reply pkg/types/vsphere/validation/platform_test.go:60]: @rvanderp3 Would you like me to still make this change?

**Comment by jcpowermac** [comment]
odd...
/test govet
/test golint
/test gofmt

**Comment by jcpowermac** [comment]
LGTM

**Comment by vr4manta** [comment]
This PR content has been moved into https://github.com/openshift/installer/pull/7179/files for feature PR

**Pull request closed without merging**